pkgs/applications/backup/timeshift/unwrapped.nix

```diff
@@ -0,0 +1,75 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, gettext
+, pkg-config
+, which
+, vala
+, gtk3
+, json-glib
+, libgee
+, vte
+, xapps
+, bash
+}:
+
+stdenv.mkDerivation rec {
+  pname = "timeshift";
+  version = "22.06.1";
+
+  src = fetchFromGitHub {
+    owner = "linuxmint";
+    repo = "timeshift";
+    rev = "v${version}";
+    sha256 = "XcxwVBKMv2YwbrI3FFWDQFs8hHruhkZq3YqzkptE6KE=";
+    fetchSubmodules = false;
+  };
+
+  patches = [
+    ./timeshift-launcher.patch
+  ];
+
+  postPatch = ''
+    find ./src -mindepth 1 -name "*.vala" -type f -exec sed -i 's/"\/sbin\/blkid"/"blkid"/g' {} \;
+    substituteInPlace ./src/makefile \
+        --replace "SHELL=/bin/bash" "SHELL=${bash}" \
+        --replace "prefix=/usr" "prefix=$out" \
+        --replace "sysconfdir=/etc" "sysconfdir=$out/etc"
```

Comment:
Could we overwrite this with makeFlags? Maybe a bit nicer than patching.

Reply:
They were hard-coded when I wrote these expressions. I'll check if things get better now.

Reply:
`prefix` and `sysconfdir` are now overridden with `makeFlagsArray` (to prevent "$out" from being escaped). `SHELL` happens to be injected by Nixpkgs automatically.

As there are still quite a few hard-coded paths in the source code, we still need to fix them in the `postPatch` phase.

By the way, the /sbin/blkid part is rewritten with `substituteInPlace` in a while loop now. The correct blkid path is injected directly (instead of using the `blkid` command from the environment) for security reason.